src/Core/Models/View/AccountView.cs

```diff
@@ -21,14 +22,21 @@ public AccountView(Account a = null, bool isActive = false)
             Email = a.Profile?.Email;
             Name = a.Profile?.Name;
             AvatarColor = a.Profile?.AvatarColor;
-            if (!string.IsNullOrWhiteSpace(a.Settings?.EnvironmentUrls?.WebVault))
-            {
-                Hostname = CoreHelpers.GetHostname(a.Settings?.EnvironmentUrls?.WebVault);
-            }
-            else if (!string.IsNullOrWhiteSpace(a.Settings?.EnvironmentUrls?.Base))
+            Hostname = ParseEndpoint(a.Settings?.EnvironmentUrls);
+        }
+
+        private string ParseEndpoint(EnvironmentUrlData urls)
+        {
+            var url = urls?.WebVault ?? urls?.Base;
+            if (!string.IsNullOrWhiteSpace(url))
             {
-                Hostname = CoreHelpers.GetHostname(a.Settings?.EnvironmentUrls?.Base);
+                if (url.Contains("bitwarden.com") || url.Contains("bitwarden.eu"))
```

Comment:
⛏️ Now that we have these constants repeated, could these be extracted to consts in `Core.Constants.cs`, like `BITWARDEN_DOMAIN_US` and `BITWARDEN_DOMAIN_EU`?
These currently exist in `HomeViewModel` but given that they're being used here as well it could be good to concentrate them in one place and reuse them.

Reply:
I'm currently working on PM-2671, refactoring the environment service, all related with this. I'm going to change this in the tech dept.